compiler-rt/lib/rtsan/rtsan_flags.cpp

```diff
@@ -0,0 +1,66 @@
+//===--- rtsan_flags.cpp - Realtime Sanitizer -------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file is a part of RealtimeSanitizer.
+//
+//===----------------------------------------------------------------------===//
+
+#include "rtsan/rtsan_flags.h"
+#include "sanitizer_common/sanitizer_flag_parser.h"
+#include "sanitizer_common/sanitizer_flags.h"
+
+using namespace __sanitizer;
+using namespace __rtsan;
+
+Flags __rtsan::flags_data;
+
+SANITIZER_INTERFACE_WEAK_DEF(const char *, __rtsan_default_options, void) {
+  return "";
+}
+
+static void RegisterRtsanFlags(FlagParser *parser, Flags *f) {
+#define RTSAN_FLAG(Type, Name, DefaultValue, Description)                      \
+  RegisterFlag(parser, #Name, Description, &f->Name);
+#include "rtsan_flags.inc"
+#undef RTSAN_FLAG
+}
+
+void Flags::SetDefaults() {
```

Comment:
Can we eliminate this function if we change line 17 of rtsan_flags.h to the following?

```cpp
#define RTSAN_FLAG(Type, Name, DefaultValue, Description) Type Name { DefaultValue };
```

Reply:
Wow, that is... extremely clever!! I had to think through what these macros are doing to figure out what was going on here.

This does work.

One possible downside of this approach, is the other sanitizers all seem to agree on the other approach, e.g.

```
> rg "define.*SAN_FLAG\(Type" -B 2
compiler-rt/lib/nsan/nsan_flags.h
16-
17-struct Flags {
18:#define NSAN_FLAG(Type, Name, DefaultValue, Description) Type Name;

compiler-rt/lib/nsan/nsan_flags.cpp
26-
27-void Flags::SetDefaults() {
28:#define NSAN_FLAG(Type, Name, DefaultValue, Description) Name = DefaultValue;

...

compiler-rt/lib/asan/asan_flags.h
30-
31-struct Flags {
32:#define ASAN_FLAG(Type, Name, DefaultValue, Description) Type Name;

compiler-rt/lib/asan/asan_flags.cpp
36-
37-void Flags::SetDefaults() {
38:#define ASAN_FLAG(Type, Name, DefaultValue, Description) Name = DefaultValue;

compiler-rt/lib/ubsan/ubsan_flags.h
22-
23-struct Flags {
24:#define UBSAN_FLAG(Type, Name, DefaultValue, Description) Type Name;

compiler-rt/lib/ubsan/ubsan_flags.cpp
34-
35-void Flags::SetDefaults() {
36:#define UBSAN_FLAG(Type, Name, DefaultValue, Description) Name = DefaultValue;
```

Whether this is by design or not, I'm not exactly sure. I want to do a second round of reviews with a broader audience on this one, possibly post-commit where we could ask a question like this.

Reply:
I have updated this to your proposed version in the latest.